src/urllib3/connection.py

```diff
@@ -741,6 +741,8 @@ def _ssl_wrap_socket_and_match_hostname(
         # `ssl` can't verify fingerprints or alternate hostnames
         assert_fingerprint
         or assert_hostname
+        # assert_hostname can be set to False to disable hostname checking
+        or assert_hostname is False
```

Comment:
Hey @pquentin , I think this check is redundant, did you mean to drop line :743 ?

``` python 
        or assert_hostname
        # assert_hostname can be set to False to disable hostname checking
        or assert_hostname is False
```

Also I tried this change originally to try and fix this issue and it added new failing tests

```
test_https_proxy_assert_hostname_non_matching
test_ssl_read_timeout
test_assert_hostname_closes_socket
```

I've not had a chance to review what the test failures are, as I agree where you're proposing this change is a more natural place for it.

Reply:
I don't think it's redundant:

 * `or assert_hostname` is for the case where assert_hostname is an actual domain
 * `or assert_hostname is False` specifically allows the False case.

We could change this to `assert_hostname is not None` but that felt less clear.

Regarding the failing tests, I'm not sure what happened, but some tests are unreliable, and they certainly seem to pass according to CI.

Reply:
Ah, I'd missed that `assert_hostname` is defaulting to `None`. Sure, agree it's more readable. And I can't reliably re-create the failing tests. Looks good 👍